src/main/java/org/dependencytrack/persistence/FindingsQueryManager.java

```diff
@@ -278,6 +281,14 @@ public List<Finding> getFindings(Project project, boolean includeSuppressed) {
                 // These are CLOB fields. Handle these here so that database-specific deserialization doesn't need to be performed (in Finding)
                 finding.getVulnerability().put("description", vulnerability.getDescription());
                 finding.getVulnerability().put("recommendation", vulnerability.getRecommendation());
+                final PackageURL purl = component.getPurl();
+                if (purl != null) {
+                    final RepositoryType type = RepositoryType.resolve(purl);
+                    if (RepositoryType.UNSUPPORTED != type) {
+                        final RepositoryMetaComponent repoMetaComponent = getRepositoryMetaComponent(type, purl.getNamespace(), purl.getName());
+                        finding.getComponent().put("latestVersion", repoMetaComponent.getLatestVersion());
+                    }
+                }
```

Comment:
It should be possible to to this via `LEFT JOIN` in the existing findings query:

https://github.com/DependencyTrack/dependency-track/blob/8f4c2b5c0d2e153363fed0ab2e865653558b7d6d/src/main/java/org/dependencytrack/model/Finding.java#L50-L81

That would save us a few database round-trips.

Reply:
The problem that I see here is that neither `COMPONENT` nor `REPOSITORY_META_COMPONENT` reference each other via a foreign key.